internal/engine/image_builder.go

```diff
@@ -3,16 +3,55 @@ package engine
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"github.com/docker/distribution/reference"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
 
 	"github.com/tilt-dev/tilt/internal/build"
 	"github.com/tilt-dev/tilt/internal/container"
 	"github.com/tilt-dev/tilt/internal/engine/buildcontrol"
 	"github.com/tilt-dev/tilt/internal/ignore"
+	"github.com/tilt-dev/tilt/pkg/logger"
 	"github.com/tilt-dev/tilt/pkg/model"
 )
 
+// Metric and label names must match the following rules:
+// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+var KeyImageRef = tag.MustNewKey("image_ref")
+
+var KeyBuildError = tag.MustNewKey("build_error")
+
+// Loosely adapted from this example:
+// https://pkg.go.dev/go.opencensus.io/plugin/ochttp
+var ImageBuildDuration = stats.Float64(
+	"image_build_duration",
+	"Image build duration",
+	stats.UnitMilliseconds)
+
+var ImageBuildDurationDistribution = view.Distribution(
+	10, 100, 500, 1000, 2000, 5000,
+	10000, 15000, 20000, 30000, 45000, 60000, 120000,
+	240000, 480000, 1000000, 2000000, 5000000)
+
+var ImageBuildDurationView = &view.View{
+	Name:        "image_build_duration_dist",
+	Measure:     ImageBuildDuration,
+	Aggregation: ImageBuildDurationDistribution,
+	Description: "Image build time, by image ref",
+	TagKeys:     []tag.Key{KeyImageRef, KeyBuildError},
+}
+
+var ImageBuildCount = &view.View{
```

Comment:
can't you just get `count` info by counting the number of histogram entries?

Reply:
this is a good question! I'm not 100% sure. i found docs that suggested you could compute counts from distributions, but also every example i saw showed them doing both.

My educated guess is that there are certain backends that don't support going from distributions -> counts